doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+
+  .. prodn::
+     ident ::= @first_letter {* @subsequent_letter }
+     first_letter ::= {| a .. z | A .. Z | _ | @unicode_letter }
+     subsequent_letter ::= {| @first_letter | @digit | ' | @unicode_id_part }
+
+  All characters are meaningful. In particular, identifiers are
+  case-sensitive.  :production:`unicode_letter` non-exhaustively
+  includes Latin, Greek, Gothic, Cyrillic, Arabic, Hebrew, Georgian,
+  Hangul, Hiragana and Katakana characters, CJK ideographs,
+  mathematical letter-like symbols and non-breaking
+  space. :production:`unicode_id_part` non-exhaustively includes
+  symbols for prime letters and subscripts.
+
+Numerals
+  Numerals are sequences of digits with an optional fractional part
+  and exponent, optionally preceded by a minus sign. :n:`@int` is an
+  integer; a numeral without fractional or exponent parts. :n:`@num`
+  is a non-negative integer.  Underscores embedded in the digits are
+  ignored, for example ``1_000_000`` is the same as ``1000000``.
+
+  .. insertprodn numeral digit
+
+  .. prodn::
+     numeral ::= {+ @digit } {? . {+ @digit } } {? {| e | E } {? {| + | - } } {+ @digit } }
+     int ::= {? - } {+ @digit }
+     num ::= {+ @digit }
+     digit ::= 0 .. 9
+
+Strings
+  Strings begin and end with ``"`` (double quote).  Use ``""`` to
+  represent a double quote character within a string.  In the grammar,
+  strings are identified with :production:`string`.
+
+Keywords
+  The following character sequences are reserved keywords that cannot
+  be used as identifiers::
+
+    _ Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop
+    SProp Set Theorem Type Variable as at by cofix discriminated else
+    end exists exists2 fix for forall fun if in lazymatch let match
+    multimatch return then using where with
+
+  Note that notations and plugins may define additional keywords.
+
+Other tokens
+  The set of tokens defined at any given time can vary because the
+  :cmd:`Notation` command can define new tokens.  A :cmd:`Require`
+  command may load more notation definitions, while the end of a
+  :cmd:`Section` may remove notations.  Some notations are defined in
+  the standard library (see :ref:`thecoqlibrary`) and are generally
+  loaded automatically at startup time.
+
+  Here are the character sequences that Coq directly defines as tokens
+  without using :cmd:`Notation`::
+
+    ! #[ % & ' ( () (bfs) (dfs) ) * ** + , - ->
+    . .( .. ... / : ::= := :> :>> ; < <+ <- <:
+    <<: <= = => > >-> >= ? @ @{ [ [= ] _
+    `( `{ { {| | |- || }
+
+  When multiple tokens match the beginning of a sequence of
+  characters, the longest matching token is used.  Occasionally you
+  may need to insert spaces to separate tokens.  For example, if ``~``
+  and ``~~`` are both defined as tokens, the inputs ``~ ~`` and ``~~``
+  generate different tokens, whereas if `~~` is not defined, then the
+  two inputs are equivalent.
+
+Basic notions
+-------------
+
+Glossary
```

Comment:
It is there because there is another sub-section at this level (Flags, Options and Tables) and I want the TOC that appears in the left-bar to be balanced.

Reply:
Why does it have to be balanced?  What difference does balancing make?  The title "Glossary" suggests like it's the one-and-only glossary in the document.

Reply:
Let's change the title then. But having only one sub-section in a section does not provide an accurate idea of the content of the section when looking at the TOC.

Reply:
You could make Flags, Options and Table a section.

I still think terms and types doesn't belong here--they're the essence of the unique functionality of Coq.  The other things in this chapter are the sort of very routine things any language manual has to cover.  Of course if you've used several proof checkers, then types and terms are routine, too, but I don't think that applies to most of our users.

Reply:
Indeed, there are good chances that a majority of our users have not used other proof assistants, however, all programming languages have "expressions" which are basically the same things as "terms" and a lot of them have "types" as well (except that they usually form a distinct class of objects.